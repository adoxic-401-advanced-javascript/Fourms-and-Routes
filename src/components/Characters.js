import React from 'react';
import PropTypes from 'prop-types';

const Characters = ({ characters }) => {

  return (
    <>
      {characters.map(({ name, status, gender, occupation, rank, image, description }) => (
        <div key={name}>
          {name && <header><h3>{name}</h3></header>}
          {image && <figure>
            <img src={image} alt={name} />
          </figure>}
          {name && status && <p>{name} is {status}</p>}
          {gender && <p>{gender}</p>}
          {occupation && <p>{occupation}</p>}
          {rank && <p>{rank}</p>}
          {description && <p>{description}</p>}
        </div>
      ))}
    </>
  );
};

Characters.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    status: PropTypes.string,
    gender: PropTypes.string,
    occupation: PropTypes.string,
    rank: PropTypes.string,
    image: PropTypes.string,
    description: PropTypes.string
  })).isRequired
};

export default Characters;
