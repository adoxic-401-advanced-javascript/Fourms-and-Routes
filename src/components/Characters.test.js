import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

describe('Characters Test', () => {
  it('Renders an Characters', () => {
    const wrapper = shallow(<Characters 
      characters={['thing']}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
