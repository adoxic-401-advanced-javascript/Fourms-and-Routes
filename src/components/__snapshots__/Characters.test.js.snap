// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters Test Renders an Characters 1`] = `
<Fragment>
  <div />
</Fragment>
`;
